ListEditor menus segfault wxPython on OS X

Running the standard ListEditor demo, I get this when modifying the list in place (ie. add item or delete item) via the ListEditor menu.  Suspect that things are not being disconnected correctly.

Traceback:
```
Exception occurred in traits notification handler.
Please check the log file for details.
Exception occurred in traits notification handler for object: <__main__.ListEditorDemo object at 0x116bf3cb0>, trait: play_list, old value: ['The Merchant of Venice', 'Hamlet', 'MacBeth'], new value: ['The Merchant of Venice', 'Hamlet']
Traceback (most recent call last):
  File "/Users/cwebster/src/ets/traits/traits/trait_notifiers.py", line 519, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "/Users/cwebster/src/ets/traits/traits/trait_notifiers.py", line 614, in dispatch
    handler( *args )
  File "/Users/cwebster/src/ets/traitsui/traitsui/editor.py", line 379, in _update_editor
    self.update_editor()
  File "/Users/cwebster/src/ets/traitsui/traitsui/wx/list_editor.py", line 186, in update_editor
    list_pane.DestroyChildren()
  File "/Users/cwebster/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/wx/_core.py", line 9191, in DestroyChildren
    return _core_.Window_DestroyChildren(*args, **kwargs)
PyAssertionError: C++ assertion "GetEventHandler() == this" failed at /Users/vagrant/pisi-64bit/tmp/wxPython-3.0.2.0-3/work/wxPython-src-3.0.2.0/src/common/wincmn.cpp(478) in ~wxWindowBase(): any pushed event handlers must have been removed
Exception occurred in traits notification handler for object: <__main__.ListEditorDemo object at 0x116bf3cb0>, trait: play_list, old value: ['The Merchant of Venice', 'Hamlet', 'MacBeth'], new value: ['The Merchant of Venice', 'Hamlet']
Traceback (most recent call last):
  File "/Users/cwebster/src/ets/traits/traits/trait_notifiers.py", line 519, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "/Users/cwebster/src/ets/traits/traits/trait_notifiers.py", line 614, in dispatch
    handler( *args )
  File "/Users/cwebster/src/ets/traitsui/traitsui/editor.py", line 379, in _update_editor
    self.update_editor()
  File "/Users/cwebster/src/ets/traitsui/traitsui/wx/list_editor.py", line 186, in update_editor
    list_pane.DestroyChildren()
  File "/Users/cwebster/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/wx/_core.py", line 9191, in DestroyChildren
    return _core_.Window_DestroyChildren(*args, **kwargs)
PyAssertionError: C++ assertion "GetEventHandler() == this" failed at /Users/vagrant/pisi-64bit/tmp/wxPython-3.0.2.0-3/work/wxPython-src-3.0.2.0/src/common/wincmn.cpp(478) in ~wxWindowBase(): any pushed event handlers must have been removed
Exception occurred in traits notification handler for object: <__main__.ListEditorDemo object at 0x116bf3cb0>, trait: play_list, old value: ['The Merchant of Venice', 'Hamlet', 'MacBeth'], new value: ['The Merchant of Venice', 'Hamlet']
Traceback (most recent call last):
  File "/Users/cwebster/src/ets/traits/traits/trait_notifiers.py", line 519, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "/Users/cwebster/src/ets/traits/traits/trait_notifiers.py", line 614, in dispatch
    handler( *args )
  File "/Users/cwebster/src/ets/traitsui/traitsui/editor.py", line 379, in _update_editor
    self.update_editor()
  File "/Users/cwebster/src/ets/traitsui/traitsui/wx/list_editor.py", line 186, in update_editor
    list_pane.DestroyChildren()
  File "/Users/cwebster/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/wx/_core.py", line 9191, in DestroyChildren
    return _core_.Window_DestroyChildren(*args, **kwargs)
PyAssertionError: C++ assertion "GetEventHandler() == this" failed at /Users/vagrant/pisi-64bit/tmp/wxPython-3.0.2.0-3/work/wxPython-src-3.0.2.0/src/common/wincmn.cpp(478) in ~wxWindowBase(): any pushed event handlers must have been removed
Traceback (most recent call last):
  File "<string>", line 2, in handler
  File "/Users/cwebster/src/ets/traitsui/traitsui/wx/list_editor.py", line 438, in delete_item
    self.update_editor()
  File "/Users/cwebster/src/ets/traitsui/traitsui/wx/list_editor.py", line 186, in update_editor
    list_pane.DestroyChildren()
  File "/Users/cwebster/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/wx/_core.py", line 9191, in DestroyChildren
    return _core_.Window_DestroyChildren(*args, **kwargs)
wx._core.PyAssertionError: C++ assertion "GetEventHandler() == this" failed at /Users/vagrant/pisi-64bit/tmp/wxPython-3.0.2.0-3/work/wxPython-src-3.0.2.0/src/common/wincmn.cpp(478) in ~wxWindowBase(): any pushed event handlers must have been removed
Segmentation fault: 11
```

--- a/traitsui/wx/image_control.py
+++ b/traitsui/wx/image_control.py
@@ -176,7 +176,7 @@ class ImageControl(wx.Window):
                 elif need_refresh:
                     self.Refresh()
                 if self._handler is not None:
-                    self._handler(self)
+                    wx.CallAfter(self._handler, self)
                 return
 
         if need_refresh:

--- a/traitsui/wx/list_editor.py
+++ b/traitsui/wx/list_editor.py
@@ -183,7 +183,8 @@ class SimpleEditor(Editor):
         # Get rid of any previous contents:
         list_pane = self.control
         list_pane.SetSizer(None)
-        list_pane.DestroyChildren()
+        for child in list_pane.GetChildren():
+            child.Destroy()
 
         # Create all of the list item trait editors:
         trait_handler = self._trait_handler
@@ -396,7 +397,7 @@ class SimpleEditor(Editor):
         item_trait = self._trait_handler.item_trait
         value = item_trait.default_value_for(self.object, self.name)
         self.value = list[:index] + [value] + list[index:]
-        self.update_editor()
+        wx.CallAfter(self.update_editor)
 
     #-------------------------------------------------------------------------
     #  Inserts a new item before the current item:
@@ -435,7 +436,7 @@ class SimpleEditor(Editor):
         """
         list, index = self.get_info()
         self.value = list[:index] + list[index + 1:]
-        self.update_editor()
+        wx.CallAfter(self.update_editor)
 
     #-------------------------------------------------------------------------
     #  Move the current item up one in the list:
@@ -447,7 +448,7 @@ class SimpleEditor(Editor):
         list, index = self.get_info()
         self.value = (list[:index - 1] + [list[index], list[index - 1]] +
                       list[index + 1:])
-        self.update_editor()
+        wx.CallAfter(self.update_editor)
 
     #-------------------------------------------------------------------------
     #  Moves the current item down one in the list:
@@ -459,7 +460,7 @@ class SimpleEditor(Editor):
         list, index = self.get_info()
         self.value = (list[:index] + [list[index + 1], list[index]] +
                       list[index + 2:])
-        self.update_editor()
+        wx.CallAfter(self.update_editor)
 
     #-------------------------------------------------------------------------
     #  Moves the current item to the top of the list:
@@ -470,7 +471,7 @@ class SimpleEditor(Editor):
         """
         list, index = self.get_info()
         self.value = [list[index]] + list[:index] + list[index + 1:]
-        self.update_editor()
+        wx.CallAfter(self.update_editor)
 
     #-------------------------------------------------------------------------
     #  Moves the current item to the bottom of the list:
@@ -481,7 +482,7 @@ class SimpleEditor(Editor):
         """
         list, index = self.get_info()
         self.value = list[:index] + list[index + 1:] + [list[index]]
-        self.update_editor()
+        wx.CallAfter(self.update_editor)
 
     #-- Property Implementations ---------------------------------------------
 

